R/function.R

```diff
@@ -0,0 +1,77 @@
+###Produce a list of award numbers in our metadata and their information
+
+#####First use download_data_objects from datamgmt
+download_data_objects <- function(mn, data_pids, out_paths, n_max = 3) {
+  stopifnot(methods::is(mn, "MNode"))
+  stopifnot(is.character(data_pids))
+  
+  for (i in seq_along(out_paths)) {
+    
+    if (file.exists(out_paths[i])) {
+      warning(call. = FALSE,
+              paste0("The file ", out_paths[i], " already exists. Skipping download."))
+    } else {
+      n_tries <- 0
+      dataObj <- "error"
+      
+      while (dataObj[1] == "error" & n_tries < n_max) {
+        dataObj <- tryCatch({
+          dataone::getObject(mn, data_pids[i])
+        }, error = function(e) {return("error")})
+        
+        n_tries <- n_tries + 1
+      }
+      writeBin(dataObj, out_paths[i])
+    }
+  }
+  
+  return(invisible())
+}
+######
+
+#Function with arguments: mn, metapid
+getAwardNumbers <- function(mn, metapid){
+  #outputs list of pid, raw_award_string, award_number, title of xml file
+  
+  stopifnot(methods::is(mn, "MNode"))
+  stopifnot(is.character(metapid))
+  
+  #get outpath name from metapid
+  systema <- getSystemMetadata(mn, metapid)
+  systema_name <- systema@fileName
+  outpath <- paste("/home/mnguyen/", systema_name, sep = "")
+  
+  #download data from metapid
+  download_data_objects(mn, c(metapid), c(outpath))
+  eml_file <- outpath
+  
+  #get xml from metapid
+  eml <- read_eml(eml_file) #read in eml_file
+  list <- list() #will be output list
+  
+  #filename 
+  filename <- eml_file
+  list[["filename"]] <- filename
+  
+  #pid
+  list[["metapid"]] <- metapid
+  
+  #the raw award string numbers
+  raw_award_string <- c()
+  funding <- eml@dataset@project@funding@para
+  for(i in 1:length(funding)){
```

Comment:
if you change this to `seq_along(funding)` then it handles the case where length = 0.  Rather than the case of `for in 1:0` which gives you an error